components/Search/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { ActionIcon, TextInput, useMantineColorScheme, useMantineTheme } from '@mantine/core';
+import { IconArrowLeft, IconArrowRight, IconSearch } from '@tabler/icons';
+import { ChangeEvent, useState } from 'react';
+import { usePropertyContext } from '../../hooks/PropertyContext';
+import style from "./_index.module.scss";
+
+function Search() {
+  const { filterProperties, properties, refreshProperties } = usePropertyContext();
+  const { colorScheme } = useMantineColorScheme();
+  const theme = useMantineTheme();
+  const [ userInput, setUserInput ] = useState<string>();
+
+  function handleSearch() {
+    if (userInput && properties) {
+      filterProperties(properties, userInput);
+    } else {
+      console.log("Please input a value");
+    }
+  }
+
+  function handleUserInput(e: ChangeEvent<HTMLInputElement>) {
+    if (e.currentTarget.value === '') {
+      refreshProperties();
+    } else {
+      setUserInput(e.currentTarget.value);
+    }
+  }
+   
+  return (
+    <TextInput
+      className={`${style.searchContainer} ${style[colorScheme]}`}
+      icon={<IconSearch size={18} stroke={2} />}
+      radius="md"
+      size="md"
+      onChange={handleUserInput}
+      rightSection={
+        <ActionIcon 
+          size={32} 
+          radius="md" 
+          color={theme.primaryColor} 
+          variant="filled" 
+          onClick={handleSearch}
+        >
+          {theme.dir === 'ltr' ? (
+            <IconArrowRight size={18} stroke={2} />
+          ) : (
+            <IconArrowLeft size={18} stroke={2} />
+          )}
```

Comment:
Place outside the return statement as a single element.

Reply:
rotate(180deg);